prime-router/src/main/kotlin/common/PhoneUtilities.kt

```diff
@@ -0,0 +1,59 @@
+package gov.cdc.prime.router.common
+
+import com.google.i18n.phonenumbers.PhoneNumberUtil
+
+/**
+ * Enumeration representing the different parts of a phone number.
+ */
+enum class PhonePart {
+    Country,
+    AreaCode,
+    Local,
+    Extension
+}
+
+/**
+ * A collection of methods for dealing with phone parts.
+ */
+object PhoneUtilities {
+
+    /**
+     *  Parses the [cleanedValue] into a PhoneNumber and returns the [part] requested.
+     *  @return either the part of the phone number requested, or null if the phone number was invalid
+     */
+    fun getPhoneNumberPart(cleanedValue: String, part: PhonePart): String? {
+        val phoneNumberUtil: PhoneNumberUtil = PhoneNumberUtil.getInstance()
+
+        // determine if this is a phone number
+        if (
+            phoneNumberUtil.isPossibleNumber(cleanedValue, "US")
+        ) {
+            val phone = phoneNumberUtil.parse(cleanedValue, "US")
+
+            // pull out correct part of the number
+            return when (part) {
+                PhonePart.Country -> phone.countryCode.toString()
+                PhonePart.AreaCode -> phone.nationalNumber.toString().substring(0, 3)
+                PhonePart.Local -> phone.nationalNumber.toString().substring(3)
+                PhonePart.Extension -> phone.extension
+            }
+        }
+        return null
+    }
+
+    /**
+     * Parses a passed in [cleanedValue] into a PhoneNumber and determines if it has an extension or not.
+     * @return a boolean indicating if the value passed in is a phone number with an extension
+     */
+    fun hasExtension(cleanedValue: String): Boolean {
```

Comment:
It is possible you do not need this function as you can just get the extension and if it is an empty value it is not set anyways